src/Avalonia.Controls/Primitives/SelectingItemsControl.cs

```diff
@@ -490,6 +508,36 @@ protected override void OnInitialized()
             }
         }
 
+        protected override void OnTextInput(TextInputEventArgs e)
+        {
+            if (!e.Handled)
+            {
+                if (!IsTextSearchEnabled)
+                    return;
+
+                StopTextSearchTimer();
```

Comment:
I would move the search logic to an asynchronous method that accepts as parameters the text to search and a CancellationToken, because if the itemcontrol contains many elements the OnTextInput event would not terminate until the search is completed.

Reply:
Wouldn't this result in OnTextInput being `async void`? Is that okay, or am I missing something? How much time should be given before the search gets cancelled?

Reply:
@workgroupengineering do you suggesting do search on thread pool? I don't think it will work nicely in general. And won't work at all, if we are doing Content?.ToString() (as it is rn), since it requires UI thread to read Content property.

Reply:
@maxkatz6 Not exactly what I wanted to suggest. I wanted to suggest queuing with low priority the search operation in the dispatcher, in order to permit the completing of  the events tunneling. If another OnTextInput is fired , the last queued search operation is cancelled and adding newone.

Reply:
I see, it might make sense. Not sure if it won't cause unwanted delay for user.